### PR TITLE
update dockerfile and build steps

### PR DESCRIPTION
**Pull request by chenzhiwei**
* remove unused steps from dockerfile
* correct the document about building docker image
* use a latest stable alpine base image
* support other architectures

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/562)

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -1,23 +1,15 @@
...
-RUN curl -fsSL -o /bin/dep https://github.com/golang/dep/releases/download/v0.5.0/dep-linux-amd64 && chmod +x /bin/dep
-
-# Download and install the latest release of dep
-ADD https://github.com/golang/dep/releases/download/v0.5.0/dep-linux-amd64 /usr/bin/dep
-RUN chmod +x /usr/bin/dep
-
 WORKDIR /go/src/github.com/kubeflow/pipelines
 COPY . .
 
-# Needed for github.com/mattn/go-sqlite3
```
Comment (IronPan) [inline backend/Dockerfile]: could you keep this comment

Comment (chenzhiwei) [reply backend/Dockerfile]: sorry, accidently removed this line.

will add it back in next pr.

**Pull request merged**